pebble/src/main/java/com/mitchellbosecke/pebble/attributes/MethodAccessValidator.java

```diff
@@ -0,0 +1,54 @@
+package com.mitchellbosecke.pebble.attributes;
+
+import com.mitchellbosecke.pebble.error.ClassAccessException;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Method;
+
+public class MethodAccessValidator {
+
+  private static final String[] FORBIDDEN_OBJECT_CLASS_METHODS = {"getClass",
+      "wait",
+      "notify",
+      "notifyAll"};
+
+  void checkIfAccessIsAllowed(Object object, Method member, String filename, int lineNumber) {
+    if (this.isAccessUnsafe(object, member)) {
+      throw new ClassAccessException(member, filename, lineNumber);
+    }
+  }
+
+  private boolean isAccessUnsafe(Object object, Method member) {
+    return object instanceof Class
+        || object instanceof Runtime
+        || object instanceof Thread
+        || object instanceof ThreadGroup
+        || object instanceof System
+        || object instanceof AccessibleObject
+        || this.isUnsafeMethodOfObjectClass(member);
+  }
+
+  private boolean isUnsafeMethodOfObjectClass(Method member) {
+    return this.isAnyOfMethods(member, "java.lang.Object", FORBIDDEN_OBJECT_CLASS_METHODS);
+  }
+
+  private boolean isAnyOfMethods(Method member, String declaringClass, String... methods) {
+    if (!this.hasDeclaringClass(member, declaringClass)) {
```

Comment:
Does checking hasDeclaringClass Object is really necessary ? If I remove this, all tests are ok

Reply:
I'd say it's not strictly needed. We could get rid of it.